lib/model/lists_database_manager.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/foundation.dart';
+import 'package:isar/isar.dart';
+import 'package:lists/model/list_model.dart';
+
+/// DatabaseManager:
+///   - a class that creates, reads, updates, and
+///     destroys lists (see ListModel).
+class ListsDatabaseManager {
+  static late final Isar isar;
+
+  static Future<void> init() async {
+    if (Isar.instanceNames.isEmpty) {
+      isar = await Isar.open([ListModelSchema], inspector: kDebugMode);
+    }
+  }
+
+  static Future<ListModel> createListModel(String name) async {
+    final newListModel = ListModel.fromTitle(name);
+    await isar.writeTxn(() async => await isar.listModels.put(newListModel));
+    return newListModel;
+  }
+
+  static Future<List<ListModel>> getAllListModels() async =>
+      await isar.listModels.where().findAll();
+
+  static Future<void> updateListModel(ListModel which) async =>
```

Comment:
My question is: why do we have to write all the list models to isar on every update? There is no way to update one specific item?

Reply:
What I had in mind was that the main function would call `getAllListModels()` and pass the returned list of `ListModel`'s to the app widget. Then, after the user selects a list, the app widget would pass the selected list to a `ListWidget`. 

So, `getAllListModels()` would be called only once at the beginning of the program (or somewhere near the beginning).

In the ListWidget, whenever the user changes the list, the updated list is written immediately to the database, but only the changed list is `put()` to the `isar.listModels` database. (In isar, each time you need to write to a database, the transactions have to be placed inside an `isar.writeTxn(() async { /* the database writes */ })` call).

So, with the current DatabaseManager, you can't _get_ a single list item, but you can _update_ a single item.

Did I answer your question?

Reply:
Kind of. From how your code reads, it looks like you are updating an _entire_ list model, not just one item. Maybe I am misunderstanding?

Reply:
Ah. You are understanding. I see what you mean. My code is updating the entire list model. 

It would be possible to update only a specific item if I tweaked the code a bit. Will do.

Reply:
Sorry, in my first answer, I mispoke. I meant to say you can't get a single list _model_, but you can update a single model.

Reply:
Ok, that makes much more sense!